stdlib/public/Cxx/CxxDictionary.swift

```diff
@@ -206,10 +206,10 @@ extension CxxDictionary {
   }
 
   @inlinable
-  public mutating func merge(
-    _ other: __owned Self,
+  public mutating func merge<T: CxxDictionary>(
+    _ other: __owned T,
     uniquingKeysWith combine: (Value, Value) throws -> Value
-  ) rethrows {
+  ) rethrows where T.Key == Key, T.Value == Value {
```

Comment:
Keep in mind this is an ABI break. Assuming that's OK, is it worth refactoring this to use typed throws instead of rethrows too?

Reply:
Good point! Having an ABI break here is fine, since the Cxx overlay isn't ABI-stable. Using the typed throws syntax would be nice!

Reply:
Done